llvm/lib/Target/AMDGPU/AMDGPULowerBufferFatPointers.cpp

```diff
@@ -2245,6 +2246,26 @@ PtrParts SplitPtrStructs::visitIntrinsicInst(IntrinsicInst &I) {
     SplitUsers.insert(&I);
     return {NewRsrc, Off};
   }
+  case Intrinsic::amdgcn_buffer_fat_ptr_load_lds: {
+    Value *BufferPtr = I.getArgOperand(0);
+    assert(isSplitFatPtr(BufferPtr->getType()) &&
+           "amdgcn.buffer.fat.pointer.load.lds has a buffer fat pointer as "
```

Comment:
Is this "has a buffer fat pointer as argument 0" an error message? I think you mean "should have ... as argument 0"?